modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/checkpoint/CheckpointHistory.java

```diff
@@ -240,21 +242,38 @@ public CheckpointEntry lastCheckpointMarkingAsInapplicable(Integer grpId) {
      * Add last checkpoint to map of the earliest checkpoints.
      *
      * @param entry Checkpoint entry.
+     * @param cacheGrpStates
      */
-    private void addCpToEarliestCpMap(CheckpointEntry entry) {
+    private void addCpToEarliestCpMap(CheckpointEntry entry, Map<Integer, CacheState> cacheGrpStates) {
```

Comment:
to avoid mistakes in further refactoring, I suggest splitting this method into two with the following signature:
* void addCpToEarliestCpMap(CheckpointEntry entry, Map<Integer, CacheState> cacheGrpStates)
* void addCpToEarliestCpMap(CheckpointEntry entry, Map<Integer, CheckpointEntry.GroupState> cacheGrpStates)

And extract following code to extra method:
```
if (!earliestCp.containsKey(grpPartKey))
         earliestCp.put(grpPartKey, entry);
```
because if tomorrow somebody changes the condition there are no guarantees that they do it in both places.

Reply:
Done